include/sparrow/buffer/buffer.hpp

```diff
@@ -151,6 +151,7 @@ namespace sparrow
         constexpr buffer(It first, It last, const A& a = A());
 
         template <std::ranges::input_range Range, allocator A = allocator_type>
+            requires std::same_as<std::ranges::range_value_t<Range>, T>
```

Comment:
Without this, we can pass a std::vector<X> to a buffer<Y>. In that case each elements are casted, with possibly some loss.